pkg/operator/k8sutil/node.go

```diff
@@ -230,19 +236,41 @@ func GetKubernetesNodesMatchingRookNodes(rookNodes []rookalpha.Node, clientset k
 
 // RookNodesMatchingKubernetesNodes returns only the given Rook nodes which have a corresponding
 // match in the list of Kubernetes nodes.
-func RookNodesMatchingKubernetesNodes(rookNodes []rookalpha.Node, kubernetesNodes []v1.Node) []rookalpha.Node {
+func RookNodesMatchingKubernetesNodes(rookStorage rookalpha.StorageScopeSpec, kubernetesNodes []v1.Node) []rookalpha.Node {
 	nodes := []rookalpha.Node{}
 	for _, kn := range kubernetesNodes {
-		for _, rn := range rookNodes {
+		for _, rn := range rookStorage.Nodes {
 			if rookNodeMatchesKubernetesNode(rn, kn) {
 				rn.Name = normalizeHostname(kn)
+
+				// modify the node Location if topology awareness enabled
+				if rookStorage.TopologyAware == true {
+					rn.Location = nodeTopologyLocation(kn, rn.Location)
+				}
+
 				nodes = append(nodes, rn)
 			}
 		}
 	}
 	return nodes
 }
 
+func nodeTopologyLocation(kubeNode v1.Node, location string) string {
+	nodeLabels := kubeNode.ObjectMeta.Labels
+	locations := []string{location}
+
+	for label := range nodeLabels {
+		for _, key := range validTopologyLabelKeys {
+			if strings.Contains(label, key) {
+				keys := strings.Split(label, "/")
+				locations = append(locations, fmt.Sprintf("%s=%s", keys[1], nodeLabels[label]))
```

Comment:
How does ceph behave when an unknown location label is specified? For example, how does the crush map learn what a `zone` and `region` are?

Reply:
@liewegas added `zone` and `region` to the crush map recently, I think. Sage, can you confirm?

Reply:
yep, https://github.com/ceph/ceph/pull/27070/files